graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx

```diff
@@ -83,70 +86,103 @@ type Props = {
 };
 
 const CachesOverview = ({ caches, pagination, paginationQueryParameter }: Props) => {
-  const { currentPage, currentPageSize, resetPage } = React.useMemo(() => ({
+  const [loading, setLoading] = React.useState(false);
+  const [localPagination, setLocalPagination] = React.useState({
     currentPage: paginationQueryParameter.page || 1,
     currentPageSize: paginationQueryParameter.pageSize || 10,
+    currentQuery: pagination.query || '',
     resetPage: paginationQueryParameter.resetPage,
-  }), [paginationQueryParameter]);
+    setPagination: paginationQueryParameter.setPagination,
+  });
+
+  React.useEffect(() => {
+    const { currentPage, currentPageSize, currentQuery } = localPagination;
+
+    LookupTableCachesActions.searchPaginated(currentPage, currentPageSize, currentQuery)
+      .then(() => setLoading(false));
+  }, [localPagination]);
 
   const onPageChange = React.useCallback((newPage: number, newPerPage: number) => {
-    LookupTableCachesActions.searchPaginated(newPage, newPerPage, pagination.query);
-  }, [pagination.query]);
+    setLocalPagination({ ...localPagination, currentPage: newPage, currentPageSize: newPerPage });
+  }, [localPagination]);
 
-  const onSearch = React.useCallback((query: string, resetLoadingStateCb: () => void) => {
-    resetPage();
-    LookupTableCachesActions.searchPaginated(currentPage, currentPageSize, query).then(resetLoadingStateCb);
-  }, [resetPage, currentPage, currentPageSize]);
+  const onSearch = React.useCallback((query: string) => {
+    localPagination.resetPage();
+    localPagination.setPagination({ page: 1, pageSize: localPagination.currentPageSize });
+    setLocalPagination({ ...localPagination, currentPage: 1, currentQuery: query });
+  }, [localPagination]);
 
   const onReset = React.useCallback(() => {
-    resetPage();
-    LookupTableCachesActions.searchPaginated(currentPage, currentPageSize);
-  }, [resetPage, currentPage, currentPageSize]);
+    localPagination.resetPage();
+    localPagination.setPagination({ page: 1, pageSize: localPagination.currentPageSize });
+    setLocalPagination({ ...localPagination, currentPage: 1, currentQuery: '' });
+  }, [localPagination]);
 
-  const emptyListComponent = pagination.query === ''
-    ? (<NoEntitiesExist>No Cache exist.</NoEntitiesExist>)
-    : (<NoSearchResult>No Cache found.</NoSearchResult>);
+  const getComponent = () => {
```

Comment:
This inline function behaves and is used like a React component. I think we should extract this into a component that is defined outside of the body of the `CachesOverview` component to make use of React's reconciliation mechanism.